### PR TITLE
[Feature] Refactor the Amazon S3 addon

### PR DESCRIPTION
**Pull request by chrisseto**
* No longer adjust Cors as its uneeded
* No longer create new osf users
* Use the key given to the osf
* Be more clear to users that listing is a required permission
* Allow multicase buckets
* Allow buckets outside of the US
* Rename API endpoints to be "standardized"
* Removed that unused API module

### CODE REVIEW

File: website/addons/s3/tests/test_view.py [hunk 1]
```diff
@@ -408,18 +356,22 @@ def test_names(self):
...
+    @mock.patch('website.addons.s3.views.crud.utils.create_bucket')
+    @mock.patch('website.addons.s3.views.crud.utils.get_bucket_names')
+    def test_create_bucket_pass(self, mock_names, mock_make):
         mock_make.return_value = True
-        mock_dropdown.return_value = ['mybucket']
-        url = self.project.api_url_for('create_new_bucket')
+        mock_names.return_value = [
+            'butintheend',
+            'it',
+            'doesntevenmatter'
```
Comment (sloria) [inline website/addons/s3/tests/test_view.py]: https://www.youtube.com/watch?v=YnFFf47bIMs

File: website/addons/s3/views/config.py [hunk 2]
```diff
@@ -1,79 +1,62 @@
...
+        raise HTTPError(httplib.BAD_REQUEST)
+
+    if not utils.can_list(access_key, secret_key):
         return {
-            'message': 'Could not access IAM. Please allow these keys permission.'
-        }, http.BAD_REQUEST
-    return {}
+            'message': ('Unable to list buckets.\n'
+                'Listing buckets is required permission that can be changed via IAM')
+        }, httplib.BAD_REQUEST
```
Comment (sloria) [inline website/addons/s3/views/config.py]: Use an `HTTPError` to return an error response and pass `message_long` to in the `data` argument.

File: website/addons/s3/views/config.py [hunk 3]
```diff
@@ -108,10 +87,10 @@ def s3_node_settings(auth, user_addon, node_addon, **kwargs):
 
     bucket = request.json.get('s3_bucket', '')
 
-    if not bucket or not does_bucket_exist(user_addon.access_key, user_addon.secret_key, bucket):
+    if not utils.bucket_exists(user_addon.access_key, user_addon.secret_key, bucket):
         error_message = ('We are having trouble connecting to that bucket. '
                          'Try a different one.')
-        return {'message': error_message}, http.BAD_REQUEST
+        return {'message': error_message}, httplib.BAD_REQUEST
```
Comment (sloria) [inline website/addons/s3/views/config.py]: Same here; use an `HTTPError`.

File: website/addons/s3/views/crud.py [hunk 4]
```diff
@@ -1,46 +1,44 @@
...
-        }
-    except S3ResponseError as e:
+        utils.create_bucket(node_addon.user_settings, bucket_name)
+    except exception.S3ResponseError as e:
         return {
             'message': e.message,
             'title': 'Problem connecting to S3',
-        }, http.NOT_ACCEPTABLE
-    except S3CreateError as e:
+        }, httplib.NOT_ACCEPTABLE
```
Comment (sloria) [inline website/addons/s3/views/crud.py]: Use `HTTPErrors` here as well.

**Comment by sloria** [comment]
Looks good. Once this is brought up to date with develop and some minor changes are made, this is good to merge. :ear_of_rice:

File: website/addons/s3/views/config.py (hunk 2 above)
Comment (chrisseto) [inline website/addons/s3/views/config.py]: I decided to keep it like this so I don't have to make any changes to the frontend code. All the inputs and outputs have been unchanged intentionally.

Comment (sloria) [inline website/addons/s3/views/config.py]: I see. That's fine. Would be good to add a TODO here then, but meh.

**Pull request merged**